cli/popper/parser.py

```diff
@@ -92,9 +94,14 @@ def complete_graph(self):
             a_block['name'] = name
 
             for n in a_block.get('needs', []):
-                if not self.workflow['action'][n].get('next', None):
-                    self.workflow['action'][n]['next'] = set()
-                self.workflow['action'][n]['next'].add(name)
+                try:
+                    if not self.workflow['action'][n].get('next', None):
+                        self.workflow['action'][n]['next'] = set()
+                    self.workflow['action'][n]['next'].add(name)
+                except KeyError:
```

Comment:
we can avoid catching the exception by doing the removal of skipped tasks to the end of the constructor (i.e. invoke `remove_skipped_actions` after `complete_graph`). When we do that, we can check whether the removal is leaving the graph in an "unrunnable" state and raise the exception notifying about it